include/xeus/xeus_context.hpp

```diff
@@ -64,25 +60,10 @@ namespace xeus
     template <class T>
     T& xcontext::get_wrapped_context()
     {
-        auto* impl = static_cast<xcontext_impl<T>*>(this);
+        auto* impl = dynamic_cast<xcontext_impl<T>*>(this);
```

Comment:
`static_cast` is fine here; the `xcontext` class is used as a "passthrough", and the end-user who instantiated it knows what it holds. So no need to check at runtime. Besides, the template class `xcontext_impl` inherits from `xcontext`, and it is unlikely that someone specializes it and breaks this hierarchy relationship.